### DISCUSSION TITLE
Getting path.split is not a function while trying to set value using Controller

### DISCUSSION THREAD

**Discussion by imsuvesh** [react-hook-form/react-hook-form#5694] [Q&A] [answered]
Requesting Help - https://stackoverflow.com/questions/68020242/getting-uncaught-typeerror-path-split-is-not-a-function-while-using-controller

CSB - https://codesandbox.io/s/react-hook-form-v7-controller-issue-6wco7
I am facing issue when I am trying to setValue inside MUI Textfields.

**Comment by JamesWong1999** [comment, marked as answer]
have a look at the doc on https://react-hook-form.com/api/useform/setvalue

`setValue(name, value)`

**Comment by thanh-nguyen-95** [comment]
Changing `setValue` to `reset` should solve the problem.

If you want to set default value for all fields, use the `defaultValues` option:

```jsx
const { register, control } = useForm({
  defaultValues: {
    Short_description: 'Subject',
    Supplier_Name: 'Sender'
  }
})
```

If you want to use `useFormContext()`, make sure to wrap your form with `FormProvider`  and passing form methods to it. See here: https://react-hook-form.com/api/useformcontext

**Comment by imsuvesh** [comment]
@JamesWong1999 Thanks for the solution but I have to write`setValue(name, value)` for each Textfield and I have 20 Textfields, I think it will make code a little dirty.
I was looking for something where I can set value of all text fields in a single line `setValue`.
I found some code on SOF which is working on older version but not in the new version, Why 🧐?
here is some sample - 
```
setValue([{Short_description: props.data.Subject},{ Supplier_Name: props.data.Sender }]);
```
and 
```
setValue({'Short_description': props.data.Subject,'Supplier_Name': props.data.Sender});
```
In both case I am getting `path.split is not a function`

 @thanh-nguyen-95 
 Thank you for the reply, but I am not trying to set `defaultValues` here, I wanted to set value in Textfield using props. For a quick csb sample, I made them as string.
 Actually, I want to setValue inside Textfield when the page loads thats why I was using useEffect.
 Ex- 
 ```
 useEffect(() => {
setValue({'Short_description': "Subject",'Supplier_Name': "Sender"});
}, []);
```

**Reply by thanh-nguyen-95** [reply]
`This function allows you to dynamically set the value of a registered field. At the same time, it tries to avoid unnecessary re-rerenders.` (https://react-hook-form.com/api/useform/setvalue)

I have no idea why you're so fixated on using `setValue`. It is clearly intended for setting value of a SINGLE field. `reset` should work in this case whether the data comes from props or some external API. For example:

```jsx
// set data from props
reset({
  textfield1: props.textfield1,
  textfield2: props.textfield2,
  textfield3: props.textfield3,
  otherField: 'foobar',
})

// set data from API
fetchYourData()
  .then(data => reset({ ...data, otherField: 'foobar' }))
  .catch(console.error)
```

If you're fetching the data first and then render the form, use the data as `defaultValues` would be a cleaner solution.

If that is not the case, please provide more information of how you structure the form so we can give you a much better answer.

**Reply by imsuvesh** [reply]
Thank You for your quick reply, Really Appreciate it. 
I am not fixated on setValue or anything, Using reset to set value is not a correct use case. If it works that doesn’t mean it's correct. Using reset to set props data undermines the actual use of reset. 

Let me explain to you what I am trying to say- 

Using `defaultValues`, I would set the form with deafult data on first load of the form. 
Using `setValue`, I would set the form data from props/api. 
Using `reset`, I would reset the form to blank fields. 

Now why I am using setValue to set props data in form because when user make changes and submits the form he would like to edit it back, at that time I won't show `defaultValues`. I would set the user form data using `setValue`.

If you still want me to give you a sample csb. Let me know

Also if this works- 
```
reset({
  textfield1: props.textfield1,
  textfield2: props.textfield2,
  textfield3: props.textfield3,
  otherField: 'foobar',
})
```
Why cant this work? It will totally solve the problem.
```
setValue({
  textfield1: props.textfield1,
  textfield2: props.textfield2,
  textfield3: props.textfield3,
  otherField: 'foobar',
})
```
I would like to see this in future improvements.
